packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/aspnet/integrationsteps.md

```diff
@@ -0,0 +1,72 @@
+### 1: Your app displays the sign-in page
+
+Create a sign-in page that captures the user's username.
+
+<div class="half border">
+
+![Screenshot showing a sign-in page with a username field and a Next button.](/img/pwd-optional/pwd-optional-sign-in-page.png)
+
+</div>
+
+> **Note**: The account's username is also its primary email address.
+
+### 2: The user submits their username
+
+When the user submits their username, create an `AuthenticationOptions` object and assign its `Username` property to the value entered by the user. Pass this object as a parameter to `IdxClient.AuthenticateAsync()`.
+
+```csharp
+var authnOptions = new AuthenticationOptions { Username = model.UserName };
+var authnResponse = await _idxClient
+   .AuthenticateAsync(authnOptions).ConfigureAwait(false);
+```
+
+### 3. The user verifies their identity with the email authenticator
+
+Query the `AuthenticationStatus` property of the `AuthenticationResponse` object returned by `AuthenticateAsync()` to discover the current status of the authentication process. If you configured your Okta org correctly, the status is `AwaitingChallengeAuthenticatorSelection`, indicating that the user needs to verify their identity and must now choose the email authenticator to complete the identity challenge.
```

Comment:
To bring the word count down somewhat, suggest rewording it with something like the following:

AuthenticateAsync() should return an AuthenticationResponse.AuthenticationStatus of AwaitingChallengeAuthenticatorSelection, which indicates the user needs to verify their identity and choose the email authenticator to complete the identity challenge.

Reply:
Changed to

`AuthenticateAsync()` returns an `AuthenticationResponse` object. Query its `AuthenticationStatus` property to discover the current status of the authentication process. A status of `AwaitingChallengeAuthenticatorSelection` indicates that the user needs to verify their identity with the email authenticator challenge.